### PR TITLE
AMI Access Control Fixes

### PR DESCRIPTION
**Pull request by alliomeria**
Changes to address https://github.com/esmero/ami/issues/151

@DiegoPino, please let me know if I missed any of the different points we discussed earlier today. Thank you!

### CODE REVIEW

File: src/Entity/Controller/amiSetEntityAccessControlHandler.php [hunk 1]
```diff
@@ -42,7 +42,7 @@ protected function checkAccess(EntityInterface $entity, $operation, AccountInter
             ->cachePerPermissions()
             ->addCacheableDependency($entity);
         }
-      case 'edit':
+      case 'update':
```
Comment (DiegoPino) [inline src/Entity/Controller/amiSetEntityAccessControlHandler.php:45]: Please keep both case statements (means don’t delete that line, the added one is good

File: src/Entity/Controller/amiSetEntityListBuilder.php [hunk 2]
```diff
@@ -67,7 +67,7 @@ public function buildRow(EntityInterface $entity) {
 
   public function getOperations(EntityInterface $entity) {
     $operations = parent::getOperations($entity); // TODO: Change the autogenerated stub
-    if ($entity->access('update') && $entity->hasLinkTemplate('process-form')) {
+    if ($entity->access('process') && $entity->hasLinkTemplate('process-form')) {
```
Comment (DiegoPino) [inline src/Entity/Controller/amiSetEntityListBuilder.php:70]: This has some of he code i shared in the issue missing. Check that one (e,g deleteados has more checks)

**Review by DiegoPino — CHANGES_REQUESTED** [review]
Hey. Thanks! Small changes needed. Added some comments. Night! (Pikachu waves)

>

File: src/Entity/Controller/amiSetEntityAccessControlHandler.php (hunk 1 above)
Comment (alliomeria) [reply src/Entity/Controller/amiSetEntityAccessControlHandler.php:45]: Ah ok, will keep both, sorry about that. (In my local Archipelago, I only have 'update' at the moment)

File: src/Entity/Controller/amiSetEntityListBuilder.php (hunk 2 above)
Comment (alliomeria) [reply src/Entity/Controller/amiSetEntityListBuilder.php:70]: Got it, will update to reference your fuller changes from the main issue. Thank you!

**Comment by alliomeria** [comment]
Thank you very much for your review and feedback, @DiegoPino! Do these small changes look ok now?

File: src/Entity/Controller/amiSetEntityAccessControlHandler.php [hunk 3]
```diff
@@ -42,7 +42,8 @@ protected function checkAccess(EntityInterface $entity, $operation, AccountInter
             ->cachePerPermissions()
             ->addCacheableDependency($entity);
         }
-      case 'edit':
+        case 'edit':
```
Comment (DiegoPino) [inline src/Entity/Controller/amiSetEntityAccessControlHandler.php:45]: Hey. This is cosmetic but helps read the code (Coding Standards). Could you indent the edit and update `case` statements the same as the `case 'view'` ?

File: src/Entity/Controller/amiSetEntityListBuilder.php [hunk 4]
```diff
@@ -67,21 +67,21 @@ public function buildRow(EntityInterface $entity) {
...
+      if ($entity->access('process') && $entity->hasLinkTemplate('report-form')) {
         $operations['reports'] = [
           'title'  => $this->t('Report'),
           'weight' => 11,
           'url'    => $this->ensureDestination($entity->toUrl('report-form')),
         ];
       }
       // If applicable to the AMI Set, add the delete processed ADOs operation.
-      if (AmiUtilityService::checkAmiSetDeleteAdosAccess($entity)) {
+      if (AmiUtilityService::checkAmiSetDeleteAdosAccess($entity) && $entity->access('deleteados') && $entity->hasLinkTemplate('delete-process-form')) {
```
Comment (DiegoPino) [inline src/Entity/Controller/amiSetEntityListBuilder.php:84]: Perfect!

**Review by DiegoPino — CHANGES_REQUESTED** [review]
Hey! Thanks so much for addressing my review. A tiny little thing more needed (Coding Standard). Added a comment. This is the last one and we merge!

File: src/Entity/Controller/amiSetEntityAccessControlHandler.php (hunk 3 above)
Comment (alliomeria) [reply src/Entity/Controller/amiSetEntityAccessControlHandler.php]: Edited--look ok now? :)

**Review by DiegoPino — APPROVED** [review]
Great!

**Comment by DiegoPino** [comment]
🥳 yay! Thanks @alliomeria

**Comment by alliomeria** [comment]
Thanks to you @DiegoPino!

**Pull request merged**